velox/core/QueryCtx.h

```diff
@@ -200,38 +206,54 @@ class ExecCtx {
     decodedVectorPool_.push_back(std::move(vector));
   }
 
-  VectorPool& vectorPool() {
-    return vectorPool_;
+  VectorPool* vectorPool() {
+    return vectorPool_.get();
   }
 
   /// Gets a possibly recycled vector of 'type and 'size'. Allocates from
   /// 'pool_' if no pre-allocated vector.
   VectorPtr getVector(const TypePtr& type, vector_size_t size) {
-    return vectorPool_.get(type, size);
+    if (vectorPool_) {
+      return vectorPool_->get(type, size);
+    } else {
+      return BaseVector::create(type, size, pool_);
+    }
   }
 
   /// Moves 'vector' to the pool if it is reusable, else leaves it in
   /// place. Returns true if the vector was moved into the pool.
   bool releaseVector(VectorPtr& vector) {
-    return vectorPool_.release(vector);
+    if (vectorPool_) {
+      return vectorPool_->release(vector);
+    }
+    return false;
   }
 
   /// Moves elements of 'vectors' to the pool if reusable, else leaves them
   /// in place. Returns number of vectors that were moved into the pool.
   size_t releaseVectors(std::vector<VectorPtr>& vectors) {
-    return vectorPool_.release(vectors);
+    if (vectorPool_) {
+      return vectorPool_->release(vectors);
+    }
+    return 0;
+  }
+
+  bool optimizeForLowMemory() const {
+    return optimizeForLowMemory_;
   }
 
  private:
   // Pool for all Buffers for this thread.
   memory::MemoryPool* pool_;
   QueryCtx* queryCtx_;
+
+  bool optimizeForLowMemory_;
```

Comment:
const and the same for pool_ and queryCtx_? Thanks!

Reply:
I'll make optimizeForLowMemory const. The other two cannot be const though because they are passed as arguments to functions that expect non-const pointers.